react-time-track/src/views/project.js

```diff
@@ -0,0 +1,178 @@
+/** @jsx jsx */
+import React from "react";
+import { jsx } from "@emotion/core";
+import Chart from "chart.js";
+import { Card, Circle } from "../components/ui";
+import { Subtitle } from "../components/ui";
+import { getProjectDetail } from "../services/project";
+import { getWeeklyReport } from "../services/weekly_report";
+import { navigate } from "@reach/router";
+
+// const weeklyData = [
+//   {
+//     id: 1,
+//     project_id: 1,
+//     week: "28",
+//     estimated_cost: 96600,
+//     real_cost: 117000
+//   },
+//   {
+//     id: 2,
+//     project_id: 1,
+//     week: "29",
+//     estimated_cost: 96600,
+//     real_cost: 117000
+//   }
+// ];
+
+const card = {
+  display: "flex",
+  justifyContent: "space-between",
+  alignItems: "center",
+  margin: "5px",
+  minWidth: "45%",
+  width: "auto"
+};
+
+function Project({ project_id }) {
+  const [project, setProject] = React.useState({ members: [] });
+  const [weeklyData, setWeeklyData] = React.useState([]);
+
+  React.useEffect(() => {
+    getProjectDetail(project_id)
+      .then(response => setProject(response))
+      .catch(error => console.log(error));
+  }, []);
+
+  React.useEffect(() => {
+    if (project.members.length == 0) return;
+    getWeeklyReport(project_id)
+      .then(response => setWeeklyData(response))
+      .catch(error => console.log(error));
+  }, [project]);
+
+  React.useEffect(() => {
+    let acumEstimated = 0;
+    let acumReal = 0;
+    const weeksLabels = weeklyData.map(week_data => week_data.week);
+    const estimated_cost = weeklyData.map(week_data => {
+      acumEstimated += week_data.estimated_cost;
+      return (acumEstimated / project.estimated_cost) * 100;
+    });
+    const real_cost = weeklyData.map(week_data => {
+      acumReal += week_data.real_cost;
+      return (acumReal / project.estimated_cost) * 100;
+    });
```

Comment:
What do you think about refactor from line 57 to 65 because you use the same function "map" and the logic is very similar among them?

Reply:
Yeah, you are right. it will be refactor on the next PR